tests/eoc_test.cpp

```diff
@@ -0,0 +1,46 @@
+#include <map>
+#include <memory>
+#include <set>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "avatar.h"
+#include "calendar.h"
+#include "cata_catch.h"
+#include "character.h"
+#include "common_types.h"
+#include "creature_tracker.h"
+#include "effect_on_condition.h"
+#include "faction.h"
+#include "field.h"
+#include "field_type.h"
+#include "game.h"
+#include "line.h"
+#include "map.h"
+#include "map_helpers.h"
+#include "memory_fast.h"
+#include "npc.h"
+#include "npc_class.h"
+#include "optional.h"
+#include "overmapbuffer.h"
+#include "pimpl.h"
+#include "player_helpers.h"
+#include "point.h"
+#include "text_snippets.h"
+#include "type_id.h"
+#include "units.h"
```

Comment:
Up to you, but these are the only headers you need here:
```suggestion
#include "avatar.h"
#include "cata_catch.h"
#include "effect_on_condition.h"
#include "map_helpers.h"
#include "point.h"
```